test/e2e/test_job_submissions.py

```diff
@@ -1727,3 +1685,59 @@ def session_action_has_expected_progress(session_action_id) -> bool:
         job.wait_until_complete(client=deadline_client)
 
         assert job.task_run_status == TaskStatus.SUCCEEDED
+
+    def test_worker_enters_stopping_state_while_draining(
+        self,
+        deadline_resources: DeadlineResources,
+        deadline_client: DeadlineClient,
+        function_worker: EC2InstanceWorker,
+        sleep_script: str = (
+            """
+            #!/usr/bin/env bash
+            while true
+            do
+                sleep 1
+            done
+            """
+            if os.environ["OPERATING_SYSTEM"] == "linux"
+            else """
+            while ($true) {{
```

Comment:
Seems like this could be a bit dangerous if the worker doesn't stop properly. THe job will run forever. Can we make this not a WHile true loop? Maybe have it timeout after 10 minutes for example.